### PR TITLE
build images for amd64 and arm64 platform (for Apple Silicon, M1, M2)

### PR DESCRIPTION
**Pull request by albertborsos**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | it builds images for amd64 and arm64

I saw it late, there is another PR (#31 ) which resolves this issue. Mine is different, because I used the native docker actions, not directly the buildx command.

The tests runs only for amd64, that is why it is fast. The master builds lasts for 1 hour both for amd64 and arm64.

Since I made it for the other repository (https://github.com/yiisoft/yii2-docker/pull/161), I updated this, with the same modifications.

Images are pushed to here: https://hub.docker.com/repository/docker/albertborsos/yii-php/general

I tested, it works.

### CODE REVIEW

**Comment by schmunk42** [comment]
> The tests runs only for amd64, that is why it is fast. The master builds lasts for 1 hour both for amd64 and arm64.

Please see https://github.com/yiisoft/yii-docker/pull/31#issuecomment-1498681445

Ideally it would build arm64 *weekly* and if we need it *manually*.

How does this PR only build on master?

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -55,7 +62,12 @@ jobs:
         docker info
         docker-compose version
     - name: Build Image
-      run: docker-compose build --build-arg X_LEGACY_GD_LIB=$X_LEGACY_GD_LIB
+      uses: docker/bake-action@v2
+      with:
+        files: docker-compose.yml
+        load: true
+        set: |
+          *.platform=linux/amd64
```
Comment (albertborsos) [inline .github/workflows/docker-image.yml:70]: That is why tests runs only for amd64, that is why it is quick for pull requests and push events.

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -70,10 +82,15 @@ jobs:
...
-    - name: Push docker image
+    - name: Build and push docker image
       if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'
-      run: docker-compose push
+      uses: docker/bake-action@v2
+      with:
+        files: docker-compose.yml
+        push: true
+        set: |
+          *.platform=linux/amd64,linux/arm64
```
Comment (albertborsos) [inline .github/workflows/docker-image.yml]: Because there is an if statement, it runs only for tags and for the master branch.

**Comment by albertborsos** [comment]
It already runs for amd64 and arm64 weekly. I do not see any advantages if it would be separated. I spent 2 days to implement this. I do not have more free time to work on this. I can develop now with my own builds, and I have to.

But you can modify this PR as you like.

**Comment by schmunk42** [comment]
I've added an additional step to build arm only on master/tags AND schedule/manually.

Thank you!

**Pull request merged**